src/sentry/api/endpoints/custom_rules.py

```diff
@@ -261,41 +267,110 @@ def _rule_to_response(rule: CustomDynamicSamplingRule) -> Response:
+
+    a string "foo" is parsed into a SearchFilter(key=SearchKey(name="message"), operator="=", value="foo")
+    we need to transform it into a SearchFilter(key=SearchKey(name="transaction"), operator="=", value="foo")
+
+    """
+    new_tokens: List[QueryToken] = []
+    for token in tokens:
+        if (
+            isinstance(token, SearchFilter)
+            and token.key.name == "message"
+            and token.operator == "="
+        ):
+            # transform the token from message to transaction
+            new_tokens.append(
+                SearchFilter(
+                    key=SearchKey("transaction"), value=token.value, operator=token.operator
+                )
+            )
+        else:
+            # nothing to change append the token as is
+            new_tokens.append(token)
+
+    return new_tokens
+
+
+def _remove_unnecessary_search_filters(tokens: Sequence[QueryToken]) -> Sequence[QueryToken]:
+    """
+    Removes unnecessary tokens that contain `event.type:x` filters.
+    """
+    new_tokens: List[QueryToken] = []
+    for token in tokens:
+        if isinstance(token, SearchFilter):
+            if token.key.name not in ["event.type"]:
+                new_tokens.append(token)
+        elif isinstance(token, ParenExpression):
+            new_tokens.append(ParenExpression(_remove_unnecessary_search_filters(token.children)))
+        else:
+            new_tokens.append(token)
+
+    return new_tokens
+
+
+def get_rule_condition(query: Optional[str]) -> RuleCondition:
+    """
+    Gets the rule condition given a query.
+
+    The rule returned, is in the format which is understood by Relay.
+    """
     try:
         if not query:
             raise UnsupportedSearchQuery(UnsupportedSearchQueryReason.NOT_TRANSACTION_QUERY)
-        else:
-            tokens = parse_search_query(query)
-
-            reason = _is_not_supported(tokens)
-            if reason is not None:
-                raise UnsupportedSearchQuery(reason)
-            # transform a simple message query into a transaction condition:
-            # "foo environment:development" -> "transaction:foo environment:development"
-            tokens = message_to_transaction_condition(tokens)
-            converter = SearchQueryConverter(tokens)
-            condition = converter.convert()
+
+        # First we parse the query.
+        tokens = cast(Sequence[QueryToken], parse_search_query(query))
+
+        # Second we check if it contains unsupported fields.
+        reason = _is_not_supported(tokens)
+        if reason is not None:
+            raise UnsupportedSearchQuery(reason)
+
+        # Third we perform transformations on the query tokens.
+        tokens = _message_to_transaction_condition(tokens)
+        tokens = _remove_unnecessary_search_filters(tokens)
+
+        # In case there are no tokens anymore, we will return a condition that always matches.
+        if not tokens:
+            return {"op": "and", "inner": []}
```

Comment:
Okay so just filtering out tokens like `event.type:transaction`, and having an empty tokens case now. 

This is more of a product question but do we make investigation rules without conditions?

Reply:
This is indeed a great question. Knowing how we designed the system I would say that also without a condition the goal would be to give users the ability to get more samples but it would generally be a super quick thing since you effectively tell Sentry "give me all my transactions until I reach 100".

I will have some chats internally with my teammates about this.